operator/controllers/btpoperator_controller.go

```diff
@@ -95,25 +103,236 @@ var (
 	labelFilter = client.MatchingLabels{labelKeyForChart: operatorName}
 )
 
-type btpOperatorGvk struct {
-	APIVersion string `yaml:"apiVersion"`
-	Kind       string `yaml:"kind"`
-}
-
 // BtpOperatorReconciler reconciles a BtpOperator object
 type BtpOperatorReconciler struct {
 	client.Client
 	*rest.Config
 	Scheme                *runtime.Scheme
+	currentVersion        string
+	wereUpdateCheckDone   bool
 	WaitForChartReadiness bool
 }
 
+func (r *BtpOperatorReconciler) handleUpdate(ctx context.Context, cr *v1alpha1.BtpOperator) error {
+	logger := log.FromContext(ctx)
+
+	configMap := r.buildBtpManagerConfigMap()
+	err := r.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
```

Comment:
you can use switch to replace if-else if chains
https://go.dev/doc/effective_go#switch